framework/src/base/Assembly.C

```diff
@@ -296,7 +371,7 @@ Assembly::reinitFE(const Elem * elem)
 
       fesd->_phi.shallowCopy(const_cast<std::vector<std::vector<Real> > &>(fe->get_phi()));
       fesd->_grad_phi.shallowCopy(const_cast<std::vector<std::vector<RealGradient> > &>(fe->get_dphi()));
-      if (_need_second_derivative[fe_type])
```

Comment:
True - that actually wasn't the problem here. It was the elem reinit too
early. However, we've fixed a number of accidental insertions lately in the
code as we try to polish several pieces of the framework. I really think we
should just avoid it across the board. It's too bad that we can't turn on
some sort of flag to trigger missing keys when we construct a std::map.
That really should be part of the standard.

On Wed, Apr 22, 2015 at 2:25 PM Derek Gaston <notifications@github.com>
wrote:

> In framework/src/base/Assembly.C
> <https://github.com/idaholab/moose/pull/4968#discussion_r28911254>:
>
> > @@ -296,7 +371,7 @@ Assembly::reinitFE(const Elem * elem)
> >
> >        fesd->_phi.shallowCopy(const_cast<std::vector<std::vector<Real> > &>(fe->get_phi()));
> >        fesd->_grad_phi.shallowCopy(const_cast<std::vector<std::vector<RealGradient> > &>(fe->get_dphi()));
> > -      if (_need_second_derivative[fe_type])
>
> I'm pretty sure you guys do plenty of that without the need for a board!
>
> What's wrong with this though? It doesn't set it true...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/4968/files#r28911254>.
>